runway/blueprints/staticsite/auth_at_edge.py

```diff
@@ -0,0 +1,416 @@
+        lambda_execution_role = self.add_lambda_execution_role()
+        check_auth_lambda = self.add_lambda(
+            'CheckAuth',
+            'Check Authorization information for request',
+            'check_auth',
+            lambda_execution_role
+        )
+        http_headers_lambda = self.add_lambda(
+            'HttpHeaders',
+            'Additional Headers added to every response',
+            'http_headers',
+            lambda_execution_role
+        )
+        parse_auth_lambda = self.add_lambda(
+            'ParseAuth',
+            'Parse the Authorization Headers/Cookies for the request',
+            'parse_auth',
+            lambda_execution_role
+        )
+        refresh_auth_lambda = self.add_lambda(
+            'RefreshAuth',
+            'Refresh the Authorization information when expired',
+            'refresh_auth',
+            lambda_execution_role
+        )
+        sign_out_lambda = self.add_lambda(
+            'SignOut',
+            'Sign the User out of the application',
+            'sign_out',
+            lambda_execution_role
+        )
+
+        # CloudFront Distribution
+        distribution_options = self.get_distribution_options(
+            bucket,
+            oai,
+            lambda_function_associations,
+            check_auth_lambda['version'],
+            http_headers_lambda['version'],
+            parse_auth_lambda['version'],
+            refresh_auth_lambda['version'],
+            sign_out_lambda['version']
+        )
+        distribution = self.add_cloudfront_distribution( # noqa pylint: disable=unused-variable
+            allow_access,
+            distribution_options
+        )
+
+    def add_lambda(self,
+                   title,  # type: str
+                   description,  # type: str
+                   handle,  # type: str
+                   role  # iam.Role
+                  ):  # noqa: E124
+        # type: (...) -> Dict[str, Any]
+        """Creates a lambda function and it's version.  # noqa
+
+        Args:
+            title (str): The name of the function in PascalCase
+            description (str): Description to be displayed in the
+                lambda panel
+            handle (str): The underscore separated representation
+                of the name of the lambda. This handle is used to
+                determine the handler for the lambda as well as
+                identify the correct Code hook_data information.
+            role (IAM.Role): The Lambda Execution Role
+        """
+        function = self.get_auth_at_edge_lambda(
+            title,
+            description,
+            handle,
+            role
+        )
+
+        return {
+            'function': function,
+            'version': self.add_version(title, function)
+        }
+
+    def add_bucket_policy(self, bucket):
+        # type: (s3.Bucket) -> s3.BucketPolicy
+        """Add a policy to the bucket if CloudFront is disabled. Ensure PublicRead.
```

Comment:
This condition ("add a policy if CloudFront is disabled") isn't correct here -- this policy is always getting applied (should probably just dedupe it with the same method in the base template).

In its use here in A@E, it should set the Principal allowed to access the bucket to the CloudFront OAI.

Reply:
Can't dedupe because the policy is different between the two blueprints. Will update the principal and the comment block.